examples/pkg/kcp/clients/listers/example/v1/clustertesttype.go

```diff
@@ -0,0 +1,108 @@
+//go:build !ignore_autogenerated
+// +build !ignore_autogenerated
+
+/*
+Copyright The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Code generated by kcp code-generator. DO NOT EDIT.
+
+package v1
+
+import (
+	kcpcache "github.com/kcp-dev/apimachinery/pkg/cache"
+	"github.com/kcp-dev/logicalcluster/v2"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/tools/cache"
+
+	examplev1 "acme.corp/pkg/apis/example/v1"
+)
+
+// ClusterTestTypeClusterLister can list ClusterTestTypes across all workspaces, or scope down to a ClusterTestTypeLister for one workspace.
+type ClusterTestTypeClusterLister interface {
+	List(selector labels.Selector) (ret []*examplev1.ClusterTestType, err error)
+	Cluster(cluster logicalcluster.Name) ClusterTestTypeLister
+}
+
+type clusterTestTypeClusterLister struct {
+	indexer cache.Indexer
+}
+
+// NewClusterTestTypeClusterLister returns a new ClusterTestTypeClusterLister.
+func NewClusterTestTypeClusterLister(indexer cache.Indexer) *clusterTestTypeClusterLister {
+	return &clusterTestTypeClusterLister{indexer: indexer}
+}
+
+// List lists all ClusterTestTypes in the indexer across all workspaces.
+func (s *clusterTestTypeClusterLister) List(selector labels.Selector) (ret []*examplev1.ClusterTestType, err error) {
+	err = cache.ListAll(s.indexer, selector, func(m interface{}) {
```

Comment:
todo: move kcp's indexers helper (using generics) into kcp apimachinery, use it here (?)